src/corelibs/U2Core/src/datatype/primers/PrimerDimersFinder.cpp

```diff
@@ -98,12 +103,16 @@ void BaseDimersFinder::fillResultsForCurrentIteration(const QByteArray& homologo
         if (!homologousRegionEnded) {
             freeEnergy += ENERGY_MAP[curArray];
         }
-        if (homologousRegionEnded || i == homologousBases.size() - 2) {
+        bool homologousBasesEnding = i == homologousBases.size() - 2;
```

Comment:
Here I fixed the but described in [UGENE-7926](https://ugene.dev/tracker/browse/UGENE-7926), which is very important for this feature